mmseg/apis/ote/apis/segmentation/inference_task.py

```diff
@@ -186,67 +188,58 @@ def hook(module, input, output):
-                dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
-
-            if dump_features:
-                for label_index, label in label_dictionary.items():
-                    if label_index == 0:
-                        continue
-
-                    if len(soft_prediction.shape) == 3:
-                        current_label_soft_prediction = soft_prediction[:, :, label_index]
-                    else:
-                        current_label_soft_prediction = soft_prediction
-
-                    min_soft_score = np.min(current_label_soft_prediction)
-                    max_soft_score = np.max(current_label_soft_prediction)
-                    factor = 255.0 / (max_soft_score - min_soft_score + 1e-12)
-                    result_media_numpy = (factor * (current_label_soft_prediction - min_soft_score)).astype(np.uint8)
-
-                    result_media = ResultMediaEntity(name=f'{label.name}',
-                                                     type='Soft Prediction',
-                                                     label=label,
-                                                     annotation_scene=dataset_item.annotation_scene,
-                                                     roi=dataset_item.roi,
-                                                     numpy=result_media_numpy)
-                    dataset_item.append_metadata_item(result_media, model=self._task_environment.model)
+        self._infer_segmentor(self._model, self._config, dataset, eval=False,
+                              output_logits=True, dump_features=True, add_predictions_to_dataset=True)
 
         pre_hook_handle.remove()
         hook_handle.remove()
 
         return dataset
-
-    @staticmethod
-    def _infer_segmentor(model: torch.nn.Module, config: Config, dataset: DatasetEntity,
+    
+    def _add_predictions_to_dataset_item(self, prediction, feature_vector, dataset_item):
+        label_dictionary = {
+            i + 1: self._labels[i] for i in range(len(self._labels))
+        }
+                
+        soft_prediction = np.transpose(prediction, axes=(1, 2, 0))
+        hard_prediction = create_hard_prediction_from_soft_prediction(
+            soft_prediction=soft_prediction,
+            soft_threshold=self._hyperparams.postprocessing.soft_threshold,
+            blur_strength=self._hyperparams.postprocessing.blur_strength,
+        )
+        annotations = create_annotation_from_segmentation_map(
+            hard_prediction=hard_prediction,
+            soft_prediction=soft_prediction,
+            label_map=label_dictionary,
+        )
+        dataset_item.append_annotations(annotations=annotations)
+        if feature_vector is not None:
+            active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
+            dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
+            for label_index, label in label_dictionary.items():
```

Comment:
Seems it's mistake - it should be enabled if `dump_features==True`.

Reply:
does `feature is not None` not guarantee that `dump_features==True`?

Reply:
Actually it depends on https://github.com/openvinotoolkit/mmsegmentation/blob/50bf757cfcf7ff07e765303d25b74794cb24cc66/mmseg/apis/ote/apis/segmentation/inference_task.py#L172-L178

Reply:
Actually it does not, `dump_features = not is_evaluation` is not used.

But anyway, does `feature is not None` not guarantee that `dump_features==True`?

Reply:
`dump_features = not is_evaluation` is not used because the related code has been removed incompletely in this PR)

Nevertheless the answer is "no it does not depend" - it's different things - features for active learning and heat maps. Unfortunately they use the local variable and method parameter with the same name `dump_features` but they are different.

Reply:
The `dump_features=True` parameter of the `_infer_segmentor` method is always `True`. But  local variable `dump_features` is depend on `is_evaluation`.

Reply:
I see, thanks

Reply:
fixed